core/Test/Tasty/Patterns.hs

```diff
@@ -28,23 +29,27 @@ import Data.Monoid
 newtype TestPattern =
   -- | @since 1.1
   TestPattern
-    (Maybe Expr)
+    [Expr]
```

Comment:
I like the idea, but I don't think you need to change the definition of `TestPattern`, this is disruptive. Instead you should be able to use `foldl' Test.Tasty.Patterns.Types.And` immediately in `optionCLParser`, combining everything into a single `Expr`.

Reply:
Good idea, that's a much smaller change.

Reply:
Mmh, somehow I liked the `[Expr]` better here (with a comment that it is to be understood conjunctively) because this better reflects what is going on here.
Of course, this would be an API change then and require a major version bump.

Reply:
Unfortunately, `[Expr]` triggers changes in many other functions throughout the package, which is way too much breakage IMO.